src/plans/D3MAavePlan.sol

```diff
@@ -0,0 +1,213 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+interface TokenLike {
+    function totalSupply() external view returns (uint256);
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface LendingPoolLike {
+    function getReserveData(address asset) external view returns (
+        uint256,    // configuration
+        uint128,    // the liquidity index. Expressed in ray
+        uint128,    // variable borrow index. Expressed in ray
+        uint128,    // the current supply rate. Expressed in ray
+        uint128,    // the current variable borrow rate. Expressed in ray
+        uint128,    // the current stable borrow rate. Expressed in ray
+        uint40,     // last updated timestamp
+        address,    // address of the adai interest bearing token
+        address,    // address of the stable debt token
+        address,    // address of the variable debt token
+        address,    // address of the interest rate strategy
+        uint8       // the id of the reserve
+    );
+}
+
+interface InterestRateStrategyLike {
+    function OPTIMAL_UTILIZATION_RATE() external view returns (uint256);
+    function EXCESS_UTILIZATION_RATE() external view returns (uint256);
+    function variableRateSlope1() external view returns (uint256);
+    function variableRateSlope2() external view returns (uint256);
+    function baseVariableBorrowRate() external view returns (uint256);
+    function getMaxVariableBorrowRate() external view returns (uint256);
+}
+
+contract D3MAavePlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateStrategyLike     public tack;
+    uint256                      public bar;   // Target Interest Rate [ray]
+
+    LendingPoolLike public immutable pool;
+    TokenLike       public immutable stableDebt;
+    TokenLike       public immutable variableDebt;
+    TokenLike       public immutable dai;
+    address         public immutable adai;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address dai_, address pool_) {
+        dai = TokenLike(dai_);
+        pool = LendingPoolLike(pool_);
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(dai_);
```

Comment:
```suggestion
        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = pool.getReserveData(dai_);
```

Reply:
This is true but less gas efficient. I guess here it doesn't matter. In general doesn't matter for the constructor.

Reply:
I've tested this locally and haven't noticed any difference in gas (`912679`).

Reply:
see https://github.com/makerdao/dss-direct-deposit/pull/78